pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -315,6 +324,23 @@ func updateFromSecret(coreClient controllerclient.Client, scope *MachineScope) e
 	scope.MachineConfig.ObjectMeta.Name = string(clusterName)
 	scope.Authorizer = authorizer
 	scope.SubscriptionID = string(subscriptionID)
+	scope.ResourceManagerEndpoint = env.ResourceManagerEndpoint
 
 	return nil
 }
+
+func (scope *MachineScope) getCloudEnvironment() (string, error) {
+	infra := &configv1.Infrastructure{}
+	infraName := controllerclient.ObjectKey{Name: globalInfrastuctureName}
+
+	if err := scope.CoreClient.Get(context.Background(), infraName, infra); err != nil {
+		return "", err
+	}
+
+	// When cloud environment is missing default to Azure Public Cloud
+	if infra.Status.PlatformStatus.Azure != nil && infra.Status.PlatformStatus.Azure.CloudName == "" {
```

Comment:
This will panic if platformstatus is nil https://github.com/openshift/cluster-api-provider-azure/pull/144#discussion_r461411556?